crash - invalid memory address or nil pointer dereference

### tlsx version:
main | dev

### Current Behavior:

```console
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1540f1b]

goroutine 83508 [running]:
github.com/projectdiscovery/tlsx/pkg/tlsx/ztls.(*Client).ConnectWithOptions(0xc0031efb78, {0xc00d0fc240, 0x10}, {0xc00d0fc240, 0x10}, {0xc00d0fc251, 0x4}, {{0x0, 0x0}, {0x0, ...}, ...})
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/tlsx@v1.0.8/pkg/tlsx/ztls/ztls.go:158 +0x5bb
github.com/projectdiscovery/tlsx/pkg/tlsx/auto.(*Client).ConnectWithOptions(0xc0031efbc0, {0xc00d0fc240, 0x10}, {0xc00d0fc240, 0x10}, {0xc00d0fc251, 0x4}, {{0x0, 0x0}, {0x0, ...}, ...})
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/tlsx@v1.0.8/pkg/tlsx/auto/auto.go:49 +0x178
github.com/projectdiscovery/tlsx/pkg/tlsx.(*Service).ConnectWithOptions(0xc0031efa70, {0xc00d0fc240, 0x10}, {0xc00d0fc240, 0x10}, {0xc00d0fc251, 0x4}, {{0x0, 0x0}, {0x0, ...}, ...})
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/tlsx@v1.0.8/pkg/tlsx/tlsx.go:74 +0x243
github.com/projectdiscovery/tlsx/pkg/tlsx.(*Service).Connect(...)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/tlsx@v1.0.8/pkg/tlsx/tlsx.go:60
github.com/projectdiscovery/nuclei/v2/pkg/protocols/ssl.(*Request).ExecuteWithResults(0xc000042c40, 0xc00d288440, 0x7f7319f4d590?, 0x7f734bb93108?, 0xc00d2ce258)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1-0.20230427224703-6321dd8a8787/pkg/protocols/ssl/ssl.go:218 +0x62d
github.com/projectdiscovery/nuclei/v2/pkg/templates.(*ClusterExecuter).Execute(0xc00540e4c0, 0xc00d2883e0)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1-0.20230427224703-6321dd8a8787/pkg/templates/cluster.go:245 +0x22d
github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeTemplateWithTargets.func2.1(0x760f26?, 0x0?, 0xc00bdedfe0)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1-0.20230427224703-6321dd8a8787/pkg/core/executors.go:128 +0x296
created by github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeTemplateWithTargets.func2
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1-0.20230427224703-6321dd8a8787/pkg/core/executors.go:105 +0x4f8
```

### Expected Behavior:
Running tlsx with no crash

### Steps To Reproduce:
....

### Anything else:
https://github.com/projectdiscovery/nuclei/issues/3619

--- a/pkg/tlsx/ztls/ztls.go
+++ b/pkg/tlsx/ztls/ztls.go
@@ -142,30 +142,26 @@ func (c *Client) ConnectWithOptions(hostname, ip, port string, options clients.C
 
 	hl := tlsConn.GetHandshakeLog()
 
-	tlsVersion := versionToTLSVersionString[uint16(hl.ServerHello.Version)]
-	tlsCipher := hl.ServerHello.CipherSuite.String()
-
-	var certificateResponse *clients.CertificateResponse
-
-	if hl != nil && hl.ServerCertificates != nil {
-		certificateResponse = ConvertCertificateToResponse(c.options, hostname, ParseSimpleTLSCertificate(hl.ServerCertificates.Certificate))
-	}
-
 	now := time.Now()
 	response := &clients.Response{
-		Timestamp:           &now,
-		Host:                hostname,
-		IP:                  resolvedIP,
-		ProbeStatus:         true,
-		Port:                port,
-		Version:             tlsVersion,
-		Cipher:              tlsCipher,
-		TLSConnection:       "ztls",
-		CertificateResponse: certificateResponse,
-		ServerName:          config.ServerName,
+		Timestamp:     &now,
+		Host:          hostname,
+		IP:            resolvedIP,
+		ProbeStatus:   true,
+		Port:          port,
+		TLSConnection: "ztls",
+		ServerName:    config.ServerName,
 	}
-	if response.CertificateResponse != nil {
-		response.Untrusted = clients.IsZTLSUntrustedCA(hl.ServerCertificates.Chain)
+	if hl != nil && hl.ServerCertificates != nil {
+		response.CertificateResponse = ConvertCertificateToResponse(c.options, hostname, ParseSimpleTLSCertificate(hl.ServerCertificates.Certificate))
+		if response.CertificateResponse != nil {
+			response.Untrusted = clients.IsZTLSUntrustedCA(hl.ServerCertificates.Chain)
+		}
+	}
+	if hl.ServerHello != nil {
+		response.Version = versionToTLSVersionString[uint16(hl.ServerHello.Version)]
+		response.Cipher = hl.ServerHello.CipherSuite.String()
+
 	}
 
 	if c.options.TLSChain {

--- a/pkg/tlsx/ztls/ztls.go
+++ b/pkg/tlsx/ztls/ztls.go
@@ -145,6 +145,12 @@ func (c *Client) ConnectWithOptions(hostname, ip, port string, options clients.C
 	tlsVersion := versionToTLSVersionString[uint16(hl.ServerHello.Version)]
 	tlsCipher := hl.ServerHello.CipherSuite.String()
 
+	var certificateResponse *clients.CertificateResponse
+
+	if hl != nil && hl.ServerCertificates != nil {
+		certificateResponse = ConvertCertificateToResponse(c.options, hostname, ParseSimpleTLSCertificate(hl.ServerCertificates.Certificate))
+	}
+
 	now := time.Now()
 	response := &clients.Response{
 		Timestamp:           &now,
@@ -155,7 +161,7 @@ func (c *Client) ConnectWithOptions(hostname, ip, port string, options clients.C
 		Version:             tlsVersion,
 		Cipher:              tlsCipher,
 		TLSConnection:       "ztls",
-		CertificateResponse: ConvertCertificateToResponse(c.options, hostname, ParseSimpleTLSCertificate(hl.ServerCertificates.Certificate)),
+		CertificateResponse: certificateResponse,
 		ServerName:          config.ServerName,
 	}
 	if response.CertificateResponse != nil {

